simplq/src/components/pages/Status/LeaveQueue.jsx

```diff
@@ -1,16 +1,23 @@
 import React from 'react';
+import CloseIcon from '@material-ui/icons/Close';
+import { useHistory } from 'react-router-dom';
 import styles from '../../../styles/statusPage.module.scss';
 
-export default () => {
+export default (props) => {
+  const history = useHistory();
+  function handleClick() {
+    props.leaveQueueHandler();
+    history.push(`/`);
```

Comment:
Check if leaveQueueHandler returns a promise, if yes, can you change to:
```
props.leaveQueueHandler().then(() => history.push(`/`));
```

Reply:
Strangely, it wasn't, even though it contains a Promise. I could fix this by adding `async` to `onDeleteClick` (the function which is called).

Reply:
inside onDeleteClick, lets add a return to line 59 [here](https://github.com/Gh05d/simplQ-frontend/blob/ticket-291-new-try/simplq/src/components/pages/Status/QueueStatus.jsx#L59)
```
return TokenService.remove().then(...).catch(...);
```

Reply:
or yeah, await on that call and make it async, upto you